streambed-confidant/src/lib.rs

```diff
@@ -183,12 +260,94 @@ impl SecretStore for FileSecretStore {
+        &self,
+        username: &str,
+        password: &str,
+    ) -> Result<UserPassAuthReply, Error> {
+        if let Ok(Some(data)) = self
+            .get_secret(&format!("auth/userpass/users/{}", username))
+            .await
+        {
+            if let Some(data_password) = data.data.data.get("password") {
+                if let Ok(data_password) = hex::decode(data_password) {
+                    let (salt, _) = data_password.split_at(SALT_SIZE);
+                    if let Ok(salt) = salt.try_into() {
+                        let password = Self::hashed(password, &salt);
+                        if password == data_password {
+                            let now = SystemTime::now();
+                            let expires = now
+                                .checked_add(USERPASS_LEASE_TIME)
+                                .unwrap_or(now)
+                                .duration_since(SystemTime::UNIX_EPOCH)
+                                .map(|t| t.as_millis())
+                                .unwrap_or(0);
+                            let data =
+                                format!(r#"{{"username":"{}","expires":{}}}"#, username, expires);
+                            let signature =
+                                hex::encode(crypto::sign(data.as_bytes(), &self.root_secret));
+                            return Ok(UserPassAuthReply {
+                                auth: AuthToken {
+                                    client_token: base64::encode(format!(
+                                        r#"{{"data":{},"signature":"{}"}}"#,
+                                        data, signature
+                                    )),
+                                    lease_duration: USERPASS_LEASE_TIME.as_secs(),
+                                },
+                            });
+                        }
+                    }
+                }
+            }
+        }
+        Err(Error::Unauthorized)
+    }
+
+    async fn token_auth(&self, token: &str) -> Result<(), Error> {
+        if let Ok(token) = base64::decode(token) {
+            if let Ok(client_token) = serde_json::from_slice::<ClientToken>(&token) {
+                let data = serde_json::to_string(&client_token.data).unwrap();
+                if let Ok(signature) = hex::decode(&client_token.signature) {
+                    if crypto::verify(data.as_bytes(), &self.root_secret, &signature) {
+                        let now = SystemTime::now()
+                            .duration_since(SystemTime::UNIX_EPOCH)
+                            .map(|t| t.as_millis())
+                            .unwrap_or(0);
+                        if now <= client_token.data.expires {
+                            return Ok(());
+                        }
+                    }
+                }
+            }
+        }
+        Err(Error::Unauthorized)
+    }
+
+    async fn userpass_create_update_user(
+        &self,
+        _current_username: &str,
```

Comment:
Should `_current_username` be an Option<String>` here? I'm passing a `""` when calling this right now

Reply:
Please pass the username in twice